tests/e2e/client/apiproxy_bookstore_quota_test.py

```diff
@@ -50,20 +50,24 @@ def verify_quota_control(self):
     response = self._call_http(path='/quota_read',
                                api_key=FLAGS.api_key)
     if response.status_code != 429:
-      while True:
+      for i in range(100):
         response = self._call_http(path='/quota_read',
                                    api_key=FLAGS.api_key)
         if response.status_code == 429:
           break;
+        elif i == 99:
+          sys.exit(utils.red("Fail to exhaust quota"))
 
     # waiting for the next quota refill.
     print("Wait for the next quota refill...");
-    while True:
+    for _ in range(100):
```

Comment:
need `i` here, thought i guess your current code will cause the job to fail anyways with a runtime error :)

Reply:
Good catch, though it hasn't been found by test due to the first quota refill protection.